src/Command/ClearOldPrivateDistsCommand.php

```diff
@@ -44,22 +40,13 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         for ($offset = 0; $offset < $count; $offset += $limit) {
             foreach ($this->query->getAllSynchronized($limit, $offset) as $package) {
-                $excludeVersion = $this->query->findLatestNonStableVersion($package->id());
-                if ($excludeVersion === null) {
-                    continue;
-                }
-
                 foreach ($this->query->findNonStableVersions($package->id()) as $version) {
-                    if ($excludeVersion->id() !== $version->id()) {
-                        $this->repository->remove(Uuid::fromString($version->id()));
-                    }
-
                     $this->packageManager->removeVersionDists(
                         $package->organization(),
                         $package->name(),
                         $version->version(),
                         'zip',
-                        $excludeVersion->reference(),
+                        $this->query->findLatestNonStableVersion($version->version())->reference(),
```

Comment:
this query is unnecessary, you can use `$version->reference()`, because the latest version is in the database

Reply:
fixed